src/client.c

```diff
@@ -1,46 +1,175 @@
 #include "client.h"
 #include <assert.h>
+#include <pthread.h>
 #include <stdlib.h>
+#include <time.h>
 
+// Diferentes niveles de bloqueo en reservas.
+static pthread_mutex_t global_mutex = PTHREAD_MUTEX_INITIALIZER;
+static pthread_mutex_t* row_mutexes;
+static pthread_mutex_t* seat_mutexes;
+
+// Inicializa Datos del cliente
 void client_data_init(client_data_t* data, movie_t* movie, int id) {
     data->movie = movie;
     data->id    = id;
 }
 
+// Inicializa el resultado de la reserva.
 void client_res_init(client_res_t* res) {
-    res->success = false; // Por defecto la reserva comienza fallando.
+    res->success = false;
 }
 
-void* client_run(client_t* client) {
-    // De momento el cliente no hace nada.
+// Puntero a función. Se asignará a la función de ejecución que le corresponda.
+void* (*client_run)(void*) = NULL;
+
+// Bloqueo Global
+void* client_global_mutex(void* arg) {
+    client_t* client = (client_t*)arg;
+    movie_t* movie   = client->data.movie;
+    int id           = client->data.id;
+    bool reserved    = false;
+
+    pthread_mutex_lock(&global_mutex); // Bloqueo global
+    int rand_row = rand() % movie->nrows;
+    int rand_col = rand() % movie->ncols;
```

Comment:
Estas dos líneas no necesitan estar en la sección crítica, muévelas antes de `pthread_mutex_lock`. Si ThreadSanitizer te da problemas, intenta utilizar `rand_r` con una variable estática, algo así

```c
static int seed = 0xdeadbeef;
int row = rand_r(&seed) & movie->nrows;
int col = rand_r(&seed) & movie->ncols;
```

Reply:
Tampoco creo que haga falta aclarar que la fila y la columna son aleatorias.
```suggestion
    int row = rand() % movie->nrows;
    int col = rand() % movie->ncols;
```